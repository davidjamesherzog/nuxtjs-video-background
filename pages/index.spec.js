import { shallowMount } from '@vue/test-utils';
import Index from '@/pages/index.vue';

describe('Index', () => {
  describe('data', () => {
    test('returns defaults', () => {
      expect(Index.data()).toEqual({
        active: false
      });
    });
  });

  describe('methods', () => {
    describe('toggleMenu', () => {
      test('should set active to true', () => {
        const mockContext = {
          active: false
        };
        Index.methods.toggleMenu.bind(mockContext)();
        expect(mockContext.active).toBeTruthy();
      });

      test('should set active to false', () => {
        const mockContext = {
          active: true
        };
        Index.methods.toggleMenu.bind(mockContext)();
        expect(mockContext.active).toBeFalsy();
      });
    });
  });

  describe('templates', () => {
    test('is a Vue instance', () => {
      const wrapper = shallowMount(Index);
      expect(wrapper.vm).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });

    test('should not add active class', () => {
      const wrapper = shallowMount(Index, {
        data: () => {
          return {
            active: false
          };
        }
      });
      const section = wrapper.find('[data-index-section]');

      expect(section.exists()).toBeTruthy();
      expect(section.classes()).toContain('showcase');
      expect(section.classes()).not.toContain('active');
      expect(wrapper.element).toMatchSnapshot();
    });

    test('should add active class', () => {
      const wrapper = shallowMount(Index, {
        data: () => {
          return {
            active: true
          };
        }
      });
      const section = wrapper.find('[data-index-section]');

      expect(section.exists()).toBeTruthy();
      expect(section.classes()).toContain('showcase');
      expect(section.classes()).toContain('active');
      expect(wrapper.element).toMatchSnapshot();
    });

    test('should validate text', () => {
      const wrapper = shallowMount(Index, {
        data: () => {
          return {
            active: false
          };
        }
      });
      const text = wrapper.find('[data-index-text]');
      const title = wrapper.find('[data-index-title]');
      const subtitle = wrapper.find('[data-index-subtitle]');
      const paragraph = wrapper.find('[data-index-paragraph]');
      const link = wrapper.find('[data-index-link]');

      expect(text.exists()).toBeTruthy();
      expect(text.classes()).toContain('text');
      expect(title.exists()).toBeTruthy();
      expect(title.text()).toContain('Make your player yours');
      expect(subtitle.exists()).toBeTruthy();
      expect(subtitle.text()).toContain(
        'most popular open source HTML5 player framework'
      );
      expect(paragraph.exists()).toBeTruthy();
      expect(paragraph.text()).toContain('Lorem ipsum dolor');
      expect(link.exists()).toBeTruthy();
      expect(link.text()).toContain('Get Started');
      expect(link.attributes().href).toEqual(
        'https://videojs.com/getting-started'
      );
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
