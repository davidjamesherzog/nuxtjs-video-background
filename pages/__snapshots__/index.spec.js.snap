// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index templates is a Vue instance 1`] = `
<div>
  <section
    class="showcase"
    data-index-section=""
  >
    <app-header-stub />
     
    <background-video-stub
      src="//vjs.zencdn.net/v/oceans.mp4"
    />
     
    <overlay-stub />
     
    <div
      class="text"
      data-index-text=""
    >
      <h2
        data-index-title=""
      >
        Make your player yours
      </h2>
       
      <h3
        data-index-subtitle=""
      >
        
        most popular open source HTML5 player framework
      
      </h3>
       
      <p
        data-index-paragraph=""
      >
        
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
        eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
        minim veniam, quis nostrud exercitation ullamco laboris nisi ut
        aliquip ex ea commodo consequat.
      
      </p>
       
      <a
        data-index-link=""
        href="https://videojs.com/getting-started"
      >
        
        Get Started
      
      </a>
    </div>
     
    <social-links-stub />
  </section>
   
  <app-menu-stub />
</div>
`;

exports[`Index templates should add active class 1`] = `
<div>
  <section
    class="showcase active"
    data-index-section=""
  >
    <app-header-stub />
     
    <background-video-stub
      src="//vjs.zencdn.net/v/oceans.mp4"
    />
     
    <overlay-stub />
     
    <div
      class="text"
      data-index-text=""
    >
      <h2
        data-index-title=""
      >
        Make your player yours
      </h2>
       
      <h3
        data-index-subtitle=""
      >
        
        most popular open source HTML5 player framework
      
      </h3>
       
      <p
        data-index-paragraph=""
      >
        
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
        eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
        minim veniam, quis nostrud exercitation ullamco laboris nisi ut
        aliquip ex ea commodo consequat.
      
      </p>
       
      <a
        data-index-link=""
        href="https://videojs.com/getting-started"
      >
        
        Get Started
      
      </a>
    </div>
     
    <social-links-stub />
  </section>
   
  <app-menu-stub />
</div>
`;

exports[`Index templates should not add active class 1`] = `
<div>
  <section
    class="showcase"
    data-index-section=""
  >
    <app-header-stub />
     
    <background-video-stub
      src="//vjs.zencdn.net/v/oceans.mp4"
    />
     
    <overlay-stub />
     
    <div
      class="text"
      data-index-text=""
    >
      <h2
        data-index-title=""
      >
        Make your player yours
      </h2>
       
      <h3
        data-index-subtitle=""
      >
        
        most popular open source HTML5 player framework
      
      </h3>
       
      <p
        data-index-paragraph=""
      >
        
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
        eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
        minim veniam, quis nostrud exercitation ullamco laboris nisi ut
        aliquip ex ea commodo consequat.
      
      </p>
       
      <a
        data-index-link=""
        href="https://videojs.com/getting-started"
      >
        
        Get Started
      
      </a>
    </div>
     
    <social-links-stub />
  </section>
   
  <app-menu-stub />
</div>
`;

exports[`Index templates should validate text 1`] = `
<div>
  <section
    class="showcase"
    data-index-section=""
  >
    <app-header-stub />
     
    <background-video-stub
      src="//vjs.zencdn.net/v/oceans.mp4"
    />
     
    <overlay-stub />
     
    <div
      class="text"
      data-index-text=""
    >
      <h2
        data-index-title=""
      >
        Make your player yours
      </h2>
       
      <h3
        data-index-subtitle=""
      >
        
        most popular open source HTML5 player framework
      
      </h3>
       
      <p
        data-index-paragraph=""
      >
        
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
        eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
        minim veniam, quis nostrud exercitation ullamco laboris nisi ut
        aliquip ex ea commodo consequat.
      
      </p>
       
      <a
        data-index-link=""
        href="https://videojs.com/getting-started"
      >
        
        Get Started
      
      </a>
    </div>
     
    <social-links-stub />
  </section>
   
  <app-menu-stub />
</div>
`;
